gormschema/gorm.go

```diff
@@ -187,3 +195,34 @@ func WithJoinTable(model any, field string, jointable any) Option {
 		})
 	}
 }
+
+type (
+	view interface {
+		ViewDef(*gorm.DB) gorm.ViewOption
+	}
+)
+
+// WithViews sets up callbacks to create views for the given "view-based" models.
+func WithViews(models ...any) Option {
+	return func(l *Loader) {
+		for _, model := range models {
+			if view, ok := model.(view); ok {
+				l.afterAutoMigrate = append(l.afterAutoMigrate, func(db *gorm.DB) error {
+					viewDef := view.ViewDef(db)
+					return db.Migrator().CreateView(inflect.Underscore(indirect(reflect.TypeOf(view)).Name()), gorm.ViewOption{
```

Comment:
Why the name is generated by provider?

Reply:
We've already named it when defining the struct, so I'll just leverage it.

Reply:
Use this to generate name: https://github.com/go-gorm/gorm/blob/master/gorm.go#L26

Reply:
Beware, there are some interface that the model can implement to provides its name:

https://github.com/go-gorm/gorm/blob/master/schema/schema.go#L107-L113

Reply:
Agreed, I've used gorm namer.TableName() to make it compatible with gorm